base/applications/msconfig/autochkpage.c

```diff
@@ -0,0 +1,247 @@
+          }
+          
+          break;
+        }
+
+        case IDC_AUTOCHK_DISABLE:
+        {
+
+          // Open the key so we can query it
+          Status = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                                SubKey,
+                                0,
+                                KEY_ALL_ACCESS,
+                                &hKey);
+
+          if (Status == ERROR_SUCCESS)
+          {
+
+            Status = RegQueryValueExW(hKey,
+                                      ValueName,
+                                      NULL,
+                                      &dwType,
+                                      NULL,
+                                      &dwcbData);
+
+            if (Status != ERROR_SUCCESS || dwType != REG_MULTI_SZ)
+            {
+              // Error, bail out
+              RegCloseKey(hKey);
+              return 1;
+            }
+
+            // Allocate a heap for the variable
+            lpsszMultiString = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, dwcbData);
+
+            Status = RegQueryValueExW(hKey,
+                                      ValueName,
+                                      NULL,
+                                      NULL,
+                                      (LPBYTE)lpsszMultiString,
+                                      &dwcbData);
+
+            if (Status != ERROR_SUCCESS)
+            {
+              // An error happened... free memory & close the key & bail out
+              HeapFree(GetProcessHeap(), 0, lpsszMultiString);
+              RegCloseKey(hKey);
+              return 1;
+            }
+
+            dwBufferSize = dwcbData;
+            ptr = lpsszMultiString;
+
+            while (*ptr)
+            {
+              szSize = wcslen(ptr) + 1;
+              if (_wcsnicmp(ptr, L"autocheck ", 10) == 0)
+              {
+                /*
+                * Found a string to remove. To do that, we just
+                * move everything that is after it backwards.
+                */
+                dwBufferSize -= szSize * sizeof(WCHAR); // Size that remains after the string to be removed.
+                memmove(ptr, ptr + szSize, dwBufferSize);
+                continue;
+              }
+ 
+              /* Otherwise, continue with the next string */
+              ptr += szSize;
+            }
+
+            // If all went good from previous points, disable autochk
+            Status = RegSetValueEx(hKey,
+                                   ValueName,
+                                   0,
+                                   REG_MULTI_SZ,
+                                   (const BYTE *)lpsszMultiString,
+                                   sizeof(dwBufferSize));
+
+            if (Status == ERROR_SUCCESS)
+            {
+              // The value set has completed successfully, autochk will be disabled
+              SendDlgItemMessage(hDlg, IDC_AUTOCHK_DISABLE, BM_SETCHECK, BST_CHECKED, 0);
+              RegCloseKey(hKey);
```

Comment:
Move the following two `RegCloseKey(hKey);` calls just before the `if (Status == ERROR_SUCCESS)`.